source/MaterialXTest/MaterialXRenderArnold/RenderArnold.cpp

```diff
@@ -76,12 +92,127 @@ bool ArnoldShaderRenderTester::runRenderer(const std::string& shaderName,
         }
     }
 
+    // Make a working copy of the document
```

Comment:
The main setup code is here along with all the preprocessing.
As noted wrapping this up into a single or set of utilities is underway. 
Some like "flattenFilenames()" is ready to use.